charms/istio-pilot/src/charm.py

```diff
@@ -181,23 +239,55 @@ def handle_ingress_auth(self, event):
 
         manifests = [rbac_configs, auth_filters]
         manifests = '\n'.join([m for m in manifests if m])
-        subprocess.run(
+        self._kubectl(
+            'delete',
+            'envoyfilters,rbacconfigs',
+            f'-lapp.juju.is/created-by={self.app.name}',
+        )
+
+        self._kubectl("apply", "-f-", input=manifests)
+
+    def _kubectl(self, *args, namespace=None, input=None, capture_output=False):
```

Comment:
`lightkube` works like a charm (eh!), see https://github.com/canonical/prometheus-k8s-operator/blob/main/lib/charms/observability_libs/v0/kubernetes_service_patch.py

Reply:
Yep!  There is a separate set of work going on to move this over to lightkube, but it is being done outside of this feature PR.